laterpay/application/Controller/Frontend/Post.php

```diff
@@ -758,4 +766,44 @@ public function ajax_load_files( LaterPay_Core_Event $event ) {
         $file_helper = new LaterPay_Helper_File();
         $file_helper->load_file( $event );
     }
+
+    /**
+     * Disable email for LaterPay enabled purchasable post type.
+     *
+     * @wp-hook jetpack_published_post_flags
+     * @param LaterPay_Core_Event $event
+     *
+     * @return void
+     */
+    public function jetpack_post_email( LaterPay_Core_Event $event ) {
+        list( $flags, $post ) = $event->get_arguments() + array( '', '' );
+
+        $is_supported_type = ( in_array( get_post_type( $post->ID ), (array) get_option( 'laterpay_enabled_post_types' ), true ) );
+
+        if ( $is_supported_type && true === LaterPay_Helper_Pricing::is_purchasable( $post->ID ) ) {
+            $flags['send_subscription'] = false;
+        }
+
+        $event->set_result( $flags );
+    }
+
+    /**
+     * Prevent the post data from being sent to the WordPress.com for LaterPay enabled purchasable post type.
+     *
+     * @wp-hook jetpack_sync_prevent_sending_post_data
+     * @param LaterPay_Core_Event $event
+     *
+     * @return void
+     */
+    public function jetpack_post_publicize( LaterPay_Core_Event $event ) {
+        list( $bool, $post ) = $event->get_arguments() + array( '', '' );
+
+        $is_supported_type = ( in_array( get_post_type( $post->ID ), (array) get_option( 'laterpay_enabled_post_types' ), true ) );
```

Comment:
Should expand this into variables? Readability concern and possible of empty array on `(array) get_option( 'laterpay_enabled_post_types' )`. @thrijith 

If possible move checking of purchasable content into a separate function.

Reply:
Or we can add more checks inside `LaterPay_Helper_Pricing::is_purchasable` itself but need to check other refs.